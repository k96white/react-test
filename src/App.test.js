import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('crash test',()=>{
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})

describe('start with count 0',()=>{
  it('count of 0', () => {
    const wrapper = shallow(<App></App>);
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 0');
  });
})

describe('on button click',()=>{
  it('count increments to 1',()=>{
    const wrapper = shallow(<App />);
    const incrBtn = wrapper.find('button.increment');
    incrBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 1');
   
  });

  it('decrements count by 1 when the decrement button is clicked', () => {
    const wrapper = shallow(<App />);
    const decrementBtn = wrapper.find('button.decrement');
    decrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: -1');
  });
})

it('matches the snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

