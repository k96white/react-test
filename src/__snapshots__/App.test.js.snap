// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Inc count
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    Decrement count
  </button>
</div>
`;
